src/main/java/network/brightspots/rcv/ResultsWriter.java

```diff
@@ -87,49 +89,131 @@ class ResultsWriter {
           StatusForRound.EXHAUSTED_CHOICE,
           StatusForRound.INVALIDATED_BY_REPEATED_RANKING);
 
+  public enum ResultType {
+    SUMMARY_REPORT("summary"/*"summary_report"*/, "csv"),
+    DETAILED_REPORT("extended_summary"/*"detailed_report"*/, "csv"),
+    JSON("summary", "json"),
+    CDF("cvr_cdf", "json"),
+    RCTAB_CVR("rctab_cvr", "csv");
 
-  // visible for testing
-  @SuppressWarnings("WeakerAccess")
-  static String sequentialSuffixForOutputPath(String sequentialTabulationId) {
-    return sequentialTabulationId != null ? "_" + sequentialTabulationId : "";
+    private final String basename;
+    private final String extension;
+
+    ResultType(String basename, String extension) {
+      this.basename = basename;
+      this.extension = extension;
+    }
+
+    public String getBasename() {
+      return basename;
+    }
+
+    public String getExtension() {
+      return extension;
+    }
   }
 
-  // visible for testing
-  @SuppressWarnings("WeakerAccess")
-  static String getOutputFilePath(
-      String outputDirectory,
-      String outputType,
-      String timestampString,
-      String sequentialTabulationId) {
-    outputType = sanitizeStringForOutput(outputType);
-    String fileName =
-        String.format(
-            "%s_%s%s",
-            timestampString, outputType, sequentialSuffixForOutputPath(sequentialTabulationId));
-    return Paths.get(outputDirectory, fileName).toAbsolutePath().toString();
+  public static class ResultFile {
+    // Since sanitizing Slice IDs can cause filename collisions, ensure each non-sanitized
+    // Slice ID is given a unique sanitized name.
+    private static final Dictionary<String, String> sliceIdToUniqueSanitizedId = new Hashtable<>();
+    // This is a set of the values in sanitizerCollisionResolution to ensure there are no collisions
+    // in the resulting sanitized names.
+    private static final HashSet<String> uniqueSanitizedIds = new HashSet<>();
+    private final ResultType resultType;
+    private final boolean isSlice;
+    private final TabulateBySlice slice;
+    private final String sliceId;
+
+    ResultFile(ResultType resultType) {
+      this.resultType = resultType;
+      this.isSlice = false;
+      slice = null;
+      sliceId = null;
+    }
+
+    ResultFile(ResultType resultType, TabulateBySlice slice, String sliceId) {
+      this.resultType = resultType;
+      this.isSlice = true;
+      this.slice = slice;
+      this.sliceId = sliceId;
+    }
+
+    // getPath helper without the modifier argument
+    public Path getPath(String directory, String prefix, Integer sequentialId) {
```

Comment:
It feels suspicious to me that the directory is a parameter to this instance method and not a field on the instance. Should I be able to pass two different directories to the same instance of this class and get two different paths?

Reply:
Yes -- that's the case for the tests, where the same object is used to find the expected and actual output files

Reply:
Created #897 to address this after 2.0